src/Canvas.tsx

```diff
@@ -270,14 +276,74 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(
       }
     }, [layout, xy]);
 
-    const dragNodeData = useMemo(
-      () =>
-        rest.dragNode
-          ? layout?.children?.find((c) => c.id === rest.dragNode.id)
-          : null,
-      [layout?.children, rest.dragNode]
+    const createDragNodeChildren = useCallback(
+      (children: any) => {
+        if (!children || !Array.isArray(children)) {
+          return [];
+        }
+
+        return children.map(({ children, ...n }) => {
+          const element =
+            typeof node === 'function' ? node(n as NodeProps) : node;
+          return (
+            <CloneElement<NodeProps>
+              key={`${n.id}-node-drag`}
+              element={element}
+              disabled
+              children={element.props.children}
+              animated={animated}
+              nodes={children}
+              childEdge={edge}
+              childNode={node}
+              {...n}
+              onDragStart={(event) => {
+                // @ts-ignore
+                setDragType(event.dragType);
+              }}
+              id={`${id}-node-${n.id}-node-drag`}
+            />
+          );
+        });
+      },
+      [animated, edge, id, node]
     );
 
+    const dragNodeData = useMemo(() => {
+      if (!rest.dragNode) {
+        return null;
+      }
+
+      const { parent } = rest.dragNode;
+      if (!parent) {
+        const foundNode = layout?.children?.find(
+          (c) => c.id === rest.dragNode.id
+        );
+        if (foundNode && foundNode.children) {
+          const nodeCopy = { ...foundNode };
+          // Node children is expecting a list of React Elements, need to create a list of elements
+          nodeCopy.children = createDragNodeChildren(nodeCopy.children);
+          // nodeCopy.children = nodeCopy.children.map(n => ({ ...n, id: `${n.id}-node-drag` }));
+          return nodeCopy;
+        }
+        return foundNode;
+      }
+
+      const foundNode = findNestedNode(
+        rest.dragNode.id,
+        layout?.children,
+        parent
+      );
+      if (foundNode && foundNode.children) {
+        const nodeCopy = { ...foundNode };
+        // Node children is expecting a list of React Elements, need to create a list of elements
+        nodeCopy.children = createDragNodeChildren(nodeCopy.children);
+        // nodeCopy.children = nodeCopy.children.map(n => ({ ...n, id: `${n.id}-node-drag` }));
+        return nodeCopy;
+      }
+
+      return foundNode;
+    }, [createDragNodeChildren, layout?.children, rest.dragNode]);
```

Comment:
Why use `rest` and not `dragNode` ?

Reply:
`rest.dragNode` uses `useCanvas` dragNode property which is `NodeData<any>` vs `dragNode` is a JSX element

Reply:
Got it - let's give it a named import then. I try to avoid rest references.